extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -233,7 +210,7 @@ public QueryDefinition makeQueryDefinition(
                            .shuffleSpec(nextShuffleSpec)
                            .processorFactory(new WindowOperatorQueryFrameProcessorFactory(
                                queryToRun,
```

Comment:
please file a PR to remove the `query` from all of these classes; they shouldn't need it - its confusing to pass that
if something is need from the `context()` flatten that into a configuration class or something and pass that

Reply:
Will take this up later as it is unrelated and shouldn't block this PR, hope that's okay.

Reply:
I see we're using 2 fields from query context (in `WindowOperatorQueryFrameProcessor`):
1. `QueryKitUtils.CTX_TIME_COLUMN_NAME` (`__timeColumn`)
2. `DruidSqlInsert.SQL_INSERT_SEGMENT_GRANULARITY` (`sqlInsertSegmentGranularity`)

Shall I go ahead and raise a PR to pass `QueryContext` instead of the query itself?

I think it will be better to pass the whole query context instead of creating a configuration class with the above 2 fields, as it would allow flexibility in the future. Thoughts?